deploy/helm/sumologic/templates/metrics/collector/otelcol/opentelemetrycollector.yaml

```diff
@@ -0,0 +1,57 @@
+{{- if and (eq (include "metrics.otelcol.enabled" .) "true") .Values.sumologic.metrics.collector.otelcol.enabled }}
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: {{ template "sumologic.metadata.name.metrics.collector.opentelemetrycollector" . }}
+  labels:
+    {{- include "sumologic.labels.metrics.opentelemetrycollector" . | nindent 4 }}
+    {{- include "sumologic.labels.common" . | nindent 4 }}
+spec:
+  mode: statefulset
+  replicas: 3
+  serviceAccount: {{ template "sumologic.metadata.name.metrics.collector.serviceaccount" . }}
+  targetAllocator:
+    serviceAccount: {{ template "sumologic.metadata.name.metrics.targetallocator.serviceaccount" . }}
+    enabled: true
+    prometheusCR:
```

Comment:
what does enabling the prometheus CR under the target allocator do?

Reply:
It makes the target allocator watch for Service and Pod Monitors, and turn them into Prometheus scrape target configuration. Without this option it only acts on the discovery config defined explicitly in prometheus receiver config.